File: leetcode/0566.Reshape-The-Matrix/ReshapeTheMatrix.go
package leetcode

// Runtime: 12 ms, faster than 96.67% of Go online submissions for Reshape the Matrix.
func matrixReshape(nums [][]int, r int, c int) [][]int {
	h := len(nums)
	w := len(nums[0])
	if h*w != r*c {
		return nums
	}

	res := make([][]int, r, r)
	for i := 0; i < r; i++ {
		res[i] = make([]int, c, c)
	}

	for i := 0; i < h*w; i++ {
		res[i/c][i%c] = nums[i/w][i%w]
	}
	return res
}

File: leetcode/0566.Reshape-The-Matrix/ReshapeTheMatrix.py
from typing import List

class Solution:

    # Runtime: 96 ms, faster than 96.03% of Python3 online submissions for Reshape the Matrix.
    def matrixReshape(self, nums: List[List[int]], r: int, c: int) -> List[List[int]]:
        h = len(nums)
        w = len(nums[0])
        if r * c != h * w: return nums

        res = [[None] * c for i in range(r)]
        for i in range(h*w):
            res[i//c][i%c] = nums[i//w][i%w]

        return res